### PR TITLE
fix: keep required after defaulting in res def

### PR DESCRIPTION
**Pull request by thxCode**
after creating a resource definition, we turn the required fields optional if we have different values.

**related**

https://github.com/seal-io/walrus/issues/2060

### CODE REVIEW

**Comment by gitlawr** [comment]
Let's add some test cases to prevent unexpected behavioral changes in the future

**Comment by thxCode** [comment]
> Let's add some test cases to prevent unexpected behavioral changes in the future

updated

File: pkg/resourcedefinitions/helper_test.go [hunk 1]
```diff
@@ -41,24 +104,69 @@ func Test_alignSchemas(t *testing.T) {
...
+				expectedOpenAPISchema.Components.Schemas == nil {
+				t.Fatalf("expected file %q has no components.schemas", expectedPath)
+			}
+
+			expectedSchemas = expectedOpenAPISchema.Components.Schemas
+		}
+
+		var nb map[string]any
+		{
+			nbJSONPath := filepath.Join(dir, tc.Name(), "nb.json")
```
Comment (gitlawr) [inline pkg/resourcedefinitions/helper_test.go:128]: What does `nb.json` stand for?

Comment (thxCode) [reply pkg/resourcedefinitions/helper_test.go:128]: notebook, a shared data uses for the previous alignSchema func, here we mock that with this json file.

Comment (gitlawr) [reply pkg/resourcedefinitions/helper_test.go:128]: Can we improve the naming or add comments

Comment (thxCode) [reply pkg/resourcedefinitions/helper_test.go:128]: comment in the testdata and the function.

**Review by gitlawr — APPROVED** [review]
LGTM

**Review by aiwantaozi — APPROVED** [review]
LGTM

**Pull request merged**